gensim/similarities/nmslib.py

```diff
@@ -0,0 +1,210 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 Radim Rehurek <me@radimrehurek.com>
+# Copyright (C) 2019 Masahiro Kazama <kazama.masa@gmail.com>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+"""
+Intro
+-----
+This module contains integration Nmslib with :class:`~gensim.models.word2vec.Word2Vec`,
+:class:`~gensim.models.doc2vec.Doc2Vec`, :class:`~gensim.models.fasttext.FastText` and
+:class:`~gensim.models.keyedvectors.KeyedVectors`.
+
+
+What is Nmslib
+-------------
+Non-Metric Space Library (NMSLIB) is an efficient cross-platform similarity search library and a toolkit
+for evaluation of similarity search methods. The core-library does not have any third-party dependencies.
+
+
+How it works
+------------
+Searching in generic non-metric space.
+
+More information about Nmslib: `github repository <https://github.com/nmslib/nmslib>`_.
+
+"""
+
+from smart_open import open
+try:
+    import cPickle as _pickle
+except ImportError:
+    import pickle as _pickle
+
+from gensim.models.doc2vec import Doc2Vec
+from gensim.models.word2vec import Word2Vec
+from gensim.models.fasttext import FastText
+from gensim.models import KeyedVectors
+from gensim.models.keyedvectors import WordEmbeddingsKeyedVectors
+try:
+    import nmslib
+except ImportError:
+    raise ImportError(
+        "Nmslib has not been installed, if you wish to use the nmslib indexer, please run `pip install nmslib`"
+    )
+
+
+class NmslibIndexer(object):
+    """This class allows to use `Nmslib <https://github.com/nmslib/nmslib>`_ as indexer for `most_similar` method
+    from :class:`~gensim.models.word2vec.Word2Vec`, :class:`~gensim.models.doc2vec.Doc2Vec`,
+    :class:`~gensim.models.fasttext.FastText` and :class:`~gensim.models.keyedvectors.Word2VecKeyedVectors` classes.
+
+    """
+
+    def __init__(self, model=None, index_params=None, query_time_params=None):
+        """
+        Parameters
+        ----------
+        model : :class:`~gensim.models.base_any2vec.BaseWordEmbeddingsModel`, optional
+            Model, that will be used as source for index.
+        index_params : dict, optional
+            index_params for Nmslib indexer.
+        query_time_params : dict, optional
+            query_time_params for Nmslib indexer.
+
+        Examples
```

Comment:
This example really belongs in the class docstring (or even better, the top-level module docstring) instead of the initializer method docstring because it demonstrates more than just the initializer method.

Reply:
Thanks. I moved this to the module docstring.